src/main/scala/com/campudus/tableaux/database/model/TableauxModel.scala

```diff
@@ -216,70 +305,81 @@ class TableauxModel(override protected[this] val connection: DatabaseConnection)
 
   private def mapRawRows(table: Table, columns: Seq[ColumnType[_]], rawRows: Seq[(RowId, Seq[Any])]): Future[Seq[Row]] = {
 
-    def mapNullValueForLinkToConcat(concatColumn: ConcatColumn, array: JsonArray): Future[List[Any]] = {
+    def mapNullValueForLinkToConcat(concatColumn: ConcatColumn, array: JsonArray): Future[List[JsonObject]] = {
       import scala.collection.JavaConverters._
 
       // Iterate over each linked row and
       // replace json's value with ConcatColumn value
-      Future.sequence(array.asScala.map({
-        case obj: JsonObject =>
-          // ConcatColumn's value is always a
-          // json array with the linked row ids
-          val rowId = obj.getLong("id").longValue()
-          retrieveCell(concatColumn, rowId).map(cell => Json.obj("id" -> rowId, "value" -> cell.value))
-      }).toList)
+      array.asScala.foldLeft(Future.successful(List.empty[JsonObject])) {
```

Comment:
Do we need to have this as a sequential future here or could we have a `Future.sequence` as I think there is only cell retrieval here?